controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +156,69 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	requiredCrds := []string{"VolumeReplicationClass", "VolumeReplication"}
+
+	for _, crd := range requiredCrds {
+		err := r.waitForResource(crd)
+		if err != nil {
+			r.Log.Error(err, "unexpected error occurred while checking CRD", "CRD", crd)
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *VolumeReplicationGroupReconciler) waitForResource(resourceName string) error {
+	result, err := r.resourceExists(resourceName)
+	if err != nil {
+		return err
+	}
+
+	if !result {
+		r.Log.Info(fmt.Sprintf("CRD %s is not installed on cluster, waiting until it appears", resourceName))
+
+		for {
+			time.Sleep(crdCheckInterval)
+
+			result, err := r.resourceExists(resourceName)
+			if err != nil {
+				return err
+			}
+
+			if result {
+				break
+			}
```

Comment:
The log "CRD % appeared on cluster" is better logged here. Otherwise if the resource
exists on the first check we will log only the second log without the first.

Reply:
Done